File: src/main/java/xyz/capybara/commands/scan/Scan.java
package xyz.capybara.commands.scan;

import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;

public class Scan extends ScanCommand {

    private String path;

    public Scan(String path) {
        this.path = path;
    }

    @Override
    public String getCommandString() {
        return "SCAN";
    }

    @Override
    protected CommandFormat getFormat() {
        return CommandFormat.NULL_CHAR;
    }

    @Override
    protected ByteBuffer getRawCommand() {
        StringBuilder rawCommand = new StringBuilder();
        rawCommand.append(getFormat().getPrefix())
                .append(getCommandString())
                .append(' ')
                .append(path)
                .append(getFormat().getTerminator());

        return ByteBuffer.wrap(rawCommand.toString().getBytes(StandardCharsets.US_ASCII));
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/scan/Scan.kt
package xyz.capybara.clamav.commands.scan

import xyz.capybara.clamav.commands.Command

import java.nio.ByteBuffer
import java.nio.charset.StandardCharsets

internal class Scan(private val path: String) : ScanCommand() {
    override val commandString
        get() = "SCAN"

    override val format
        get() = CommandFormat.NULL_CHAR

    override val rawCommand: ByteBuffer
        get() = ByteBuffer.wrap("${format.prefix}$commandString $path${format.terminator}".toByteArray(StandardCharsets.US_ASCII))
}
